sql/plan/recursive_cte.go

```diff
@@ -43,7 +43,7 @@ import (
 // projection count and types. [Init] will be resolved before
 // [Rec] or [RecursiveCte] to share schema types.
 type RecursiveCte struct {
-	union *Union
+	setOp *SetOp
```

Comment:
RecursiveCTEs can only be unions, right? So the names in this file should probably reflect that. (And we should make sure that the SetOp has its op type set to Union.